### ISSUE TITLE
deno: command not found when running from git-for-windows shell

### ISSUE THREAD

**Issue by CodeWitchBella** [denoland/deno_task_shell#27]
Contents of deno.json

```json
{
    "tasks": {
        "version": "deno --version"
    }
}
```

I run the task by running `deno task version` and get

```
Warning deno task is unstable and may drastically change in the future
Task version deno --version
deno: command not found
```

Expected output

```
Warning deno task is unstable and may drastically change in the future
Task version deno --version
deno 1.20.3 (release, x86_64-pc-windows-msvc)
v8 10.0.139.6
typescript 4.6.2
```

To workaround this issue I can set the task to include `deno.exe` instead of `deno`, but then it stops working in powershell and non-windows platforms. Please let me know if you need more information to reproduce this issue 🙂

**Comment by dsherret** [comment]
Hmmm... strange. What's the output of running `deno task output` with this deno.json?

```json
{
  "tasks": {
    "output": "echo $PATHEXT"
  }
}
```

Also, if you do the following, does it work?

```json
{
    "tasks": {
        "version": "deno.EXE --version"
    }
}
```

I'm wondering if perhaps the directory containing the executable is case sensitive and so it finds `deno.exe`, but not `deno.EXE` (I just learned that Windows can set case sensitivey for a directory... your file system is NTFS, correct?)

**Comment by dsherret** [comment]
Also, what's the output of `where deno` in cmd?

**Comment by CodeWitchBella** [comment]
where deno: `C:\Users\Isabella Skořepová\.deno\bin\deno.exe`

with `deno.EXE`: ✅ yes, it works

PATHEXT:
```
Task version echo $PATHEXT
.COM;.EXE;.BAT;.CMD;.VBS;.VBE;.JS;.JSE;.WSF;.WSH;.MSC;.PY;
```

Yes, my filesystem is case-insensitive NTFS

**Comment by CodeWitchBella** [comment]
More potentially useful information:

```bash
$ echo $PATH
/mingw64/bin:/usr/bin:/c/Users/Isabella Skořepová/bin:/c/Program Files/Oculus/Support/oculus-runtime:/c/Python39/Scripts:/c/Python39:/c/Windows/system32:/c/Windows:/c/Windows/System32/Wbem:/c/Windows/System32/WindowsPowerShell/v1.0:/c/Windows/System32/OpenSSH:/c/Program Files (x86)/NVIDIA Corporation/PhysX/Common:/c/ProgramData/chocolatey/bin:/c/Stuff/WinSCP:/c/Stuff/gnuplot/bin:/c/Program Files/nodejs:/cmd:/c/Program Files/dotnet:/c/Users/Isabella Skořepová/AppData/Local/Programs/Microsoft VS Code/bin:/c/Users/Isabella Skořepová/.dotnet/tools:/c/tools/gsutil:/c/Users/Isabella Skořepová/AppData/Local/Microsoft/WindowsApps:/c/Users/Isabella Skořepová/.dotnet/tools:/c/Users/Isabella Skořepová/.deno/bin:/c/Users/Isabella Skořepová/AppData/Roaming/npm:/c/Users/Isabella Skořepová/.dotnet/tools:'C:\Program Files\Microsoft Visual Studio\2022\Community\Common7\IDE\CommonExtensions\Microsoft\CMake\Ninja\'

$ which deno
/c/Users/Isabella Skořepová/.deno/bin/deno

$ ls '/c/Users/Isabella Skořepová/.deno/bin'
deno.exe  deno.old.exe
```

**Comment by dsherret** [comment]
Oh! Does this use colon separators instead of semi-colon?

```json
{
    "tasks": {
        "output": "echo $PATH"
    }
}
```

**Comment by CodeWitchBella** [comment]
```
$ deno task output
Warning deno task is unstable and may drastically change in the future
Task output echo $PATH
C:\Stuff\Git\mingw64\bin;C:\Stuff\Git\usr\bin;C:\Users\Isabella Sko┼Öepov├í\bin;C:\Program Files\Oculus\Support\oculus-runtime;C:\Python39\Scripts;C:\Python39;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0;C:\Windows\System32\OpenSSH;C:\Program Files (x86)\NVIDIA Corporation\PhysX\Common;C:\ProgramData\chocolatey\bin;C:\Stuff\WinSCP;C:\Stuff\gnuplot\bin;C:\Program Files\nodejs;C:\Stuff\Git\cmd;C:\Program Files\dotnet;C:\Users\Isabella Sko┼Öepov├í\AppData\Local\Programs\Microsoft VS Code\bin;C:\Users\Isabella Sko┼Öepov├í\.dotnet\tools;C:\tools\gsutil;C:\Users\Isabella Sko┼Öepov├í\AppData\Local\Microsoft\WindowsApps;C:\Users\Isabella Sko┼Öepov├í\.dotnet\tools;C:\Users\Isabella Sko┼Öepov├í\.deno\bin;C:\Users\Isabella Sko┼Öepov├í\AppData\Roaming\npm;C:\Users\Isabella Sko┼Öepov├í\.dotnet\tools;'C;C:\Program Files\Microsoft Visual Studio\2022\Community\Common7\IDE\CommonExtensions\Microsoft\CMake\Ninja\'
```

It outputs semicolons but also corrupts the ř in my name.

**Comment by dsherret** [comment]
@CodeWitchBella thanks for all the information. I think I found the issue and am able to reproduce it. Try running this in your terminal and afterwards see if the original command works:

```shell
export PATHEXT=".COM;.EXE;.BAT;.CMD;.VBS;.VBE;.JS;.JSE;.WSF;.WSH;.MSC;.PY"
```

The problem was with the trailing semi-colon in PATHEXT. There's an `ends_with(ext)` check done here and when there was a blank ext in PATHEXT it would always return true:

https://github.com/denoland/deno_task_shell/blob/b7aef06ccac249af346dbf510037e955d58670bf/src/shell.rs#L456-L457

It seems my fix in #28 fixes this. I'll update that PR to handle it.

Also, interestingly, I'm not able to reproduce the issue you were seeing with your name being like that.

**Comment by CodeWitchBella** [comment]
Changing PATHEXT as you suggested indeed fixes the issue 🎉 

Thanks for the fast response 🙂 

I wouldn't worry about the encoding issue in the name - when I tried it in windows sandbox it also shows up correctly so it probably is some other weirdness in my env. Given that it works, it's not a big problem.
